src/lib/swapper/swappers/OsmosisSwapper/OsmosisSwapper.ts

```diff
@@ -147,16 +146,40 @@ export class OsmosisSwapper implements Swapper<ChainId> {
         }),
       )
 
+    const sellAssetIsOnOsmosisNetwork = sellAsset.chainId === osmosisChainId
+    const cosmosFees = await cosmosAdapter.getFeeData(getFeeDataInput)
+    const initiatingTxFeeData = sellAssetIsOnOsmosisNetwork ? osmosisFees : cosmosFees
+
     return Ok({
       buyAmountBeforeFeesCryptoBaseUnit: buyAmountCryptoBaseUnit,
       buyAsset,
       feeData: {
         networkFeeCryptoBaseUnit: fee,
         protocolFees: {
-          [buyAsset.assetId]: {
-            amountCryptoBaseUnit: buyAssetTradeFeeCryptoBaseUnit,
-            requiresBalance: true,
-            asset: buyAsset,
+          // Note, the current implementation is a hack where we consider the whole swap as one hop
+          // This is only there to make the fees correct in the UI, but this isn't a "protocol fee", it's a network fee for the second hop (the IBC transfer)
+          ...(sellAssetIsOnOsmosisNetwork
+            ? {
+                [buyAsset.assetId]: {
```

Comment:
this should be osmosis "protocol" fee to represent the second hop tx (msg_transfer) on osmosis

Reply:
then there would be the addition of the atom protocol fee (swap_exact_amount_in) as:
```
{
              [buyAsset.assetId]: {
                  amountCryptoBaseUnit: buyAssetTradeFeeCryptoBaseUnit,
                  requiresBalance: false,
                  asset: buyAsset,
              },
}
```